reflection.md

```diff
@@ -2,17 +2,51 @@
 
 ## Which operators were easiest to implement and why?
 
+I found operators `||` and `~` the easiset to implement, since they were
+essentially just functions which take another RegularExpression.
+
+The operators `<*>` and `<+>` were slightly more interesting, since it
+required an investigation into how to define a postfix operator. I originally
+thought it would require using a variant of the `unary_` modifier used for
+prefix unary operators, but it turns out that the `unary_` modifier is only
+used to distinguish between pre and post fix unary operators.
```

Comment:
Oh, I didn't know that (actually I had completely forgotten about `unary_`). Good to know.